sdks/python/apache_beam/io/filebasedsource_test.py

```diff
@@ -67,9 +67,7 @@ def read_records(self, file_name, range_tracker):
         start += len(line)
       current = start
       line = f.readline()
-      while line:
-        if not range_tracker.try_claim(current):
-          return
+      while range_tracker.try_claim(current):
```

Comment:
I think the try_claim has handled the empty lime. When reading the last line, the try_claim will fail because of out_of_range, thus the last empty line will not be emitted.

Reply:
I saw what you mean now. The range_tracker may end with INF when the source is not splittable, where range_tracker.try_claim() is not enough to indicate whether reaching the end of the file.

Reply:
Fixed in commit https://github.com/apache/beam/pull/9058/commits/7db08dbfeef1a4e6785a8d20816e35deb747871a